platform/ios/src/MGLMapView.mm

```diff
@@ -942,6 +942,23 @@ - (void)layoutSubviews
     [self updateUserLocationAnnotationView];
 
     [self updateAttributionAlertView];
+    
+    if (!CGRectContainsRect(self.bounds, self.attributionButton.mgl_frameForTransformIdentity)) {
```

Comment:
Can we throw the exceptions from where the user sets the positions & margins? There's no (easy) way for the user to catch these exceptions (other than subclassing `MGLMapView`) - so throwing them from where the user sets them would be ideal (though I appreciate that the view may not yet be set up 🤔).

Reply:
Hmm. Given that the developer can't catch this exception, I think we should convert these to `MGLAssert` - that will log a "Fault" level message, and then assert.

So we'll hit the assert in debug, but the devs can see a log message.